google/cloud/storage/_helpers.py

```diff
@@ -581,3 +581,20 @@ def _api_core_retry_to_resumable_media_retry(retry, num_retries=None):
         return resumable_media.RetryStrategy(max_retries=num_retries)
     else:
         return resumable_media.RetryStrategy(max_retries=0)
+
+
+def _get_default_headers(user_agent):
+    """Get the headers for a request.
+
+    Args:
+        user_agent (str): The user-agent for requests.
+    Returns:
+        Dict: The headers to be used for the request.
+    """
+    return {
+        "Accept": "application/json",
+        "Accept-Encoding": "gzip, deflate",
+        "User-Agent": user_agent,
+        "X-Goog-Api-Client": user_agent,
```

Comment:
This is completely replacing x-goog-api-client; is it possible to keep the existing x-goog-api-client header value while adding `gcloud-python/`?

Reply:
Hmm, I'm not necessarily sure it is. This helper alone isn't really modifying any headers, as much as it's just setting the default. I think the thing to here though would be in the two places this is called, rearrange the order of the header dicts, so that the default one comes first which means anything specified in the other header will override these.

Reply:
Ohhh although perhaps somewhere later this would override anything happening. I admittedly don't know much about X-Goog-Api-Client, I just know that @andrewsg suggested we set that as well in this change.

Reply:
I see you started adding tests, I think that may clarify if we need to care about information passed through x-goog-api-client vs. User-Agent.